servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -295,6 +295,14 @@ boolean persistFileInternal(long fileId, long length, long opTimeMs)
       file.setPersisted(true);
       needLog = true;
     }
+    // Recursively traverse up the tree and mark all parent directories as persisted as well
+    long parentId = file.getParentId();
+    while (parentId != InodeTree.NO_PARENT) {
+      Inode parent = mInodeTree.getInodeById(parentId);
```

Comment:
Yeah, this should not be re-implemented here. You should use propagatePersisted()